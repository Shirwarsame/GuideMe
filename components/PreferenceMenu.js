import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Icon, Button } from "native-base";

/**
 * US12 - As a user, I want to be able to select a destination building by clicking on it.
 * US14 - As a user, I should be able to set my current location as the starting point.
 * US18 - As a user, I should be able to choose “walking” as a means of transportation.
 * US19 - As a user, I should be able to choose public transport as a means of transportation.
 * US20 - As a user, I should be able to choose my car as a means of transportation.
 * US21 - As a user, I should be able to choose Concordia Shuttle as a means of transportation.
 *
 * The following function renders a preference menu with button for the user to select
 * his method of transport and so on.
 */

//TODO: Link this file to the PreviewDirection.js

function PreferenceMenu () {
    // const [userType, setUserType] = React.useState("");
    // const [mobilityReduced, setMobilityReduced] = React.useState(false);
    // const [mobilityNotReduced, setMobilityNotReduced] = React.useState(false);
    // const [travelType, setTravelType] = React.useState(false);

    return (
        <View style={styles.container}>
            <Text style={styles.mainLabel}>Preferences</Text>

            <View style={styles.containerOfButtons1}>
                <View style={styles.labelContainer}>
                    <Text style={styles.shortLabel}>I am: </Text>
                </View>
                <Button transparent style={styles.buttonContainer}>
                    <Text style={styles.buttonLabel}>Graduate Student</Text>
                </Button>
                <Button transparent style={styles.buttonContainer}>
                    <Text style={styles.buttonLabel}>Undergrad Student</Text>
                </Button>
                <Button transparent style={styles.buttonContainer}>
                    <Text style={styles.buttonLabel}>Visitor</Text>
                </Button>
                <Button transparent style={styles.buttonContainer}>
                    <Text style={styles.buttonLabel}>University Staff</Text>
                </Button>
            </View>

            <View style={styles.containerOfButtons2}>
                <View style={styles.labelContainer}>
                    <Text style={styles.shortLabel}>Mobility Reduced: </Text>
                </View>
                <Button transparent style={styles.buttonContainer} >
                    <Text style={styles.buttonLabelMobility} > Yes </Text>
                </Button>
                <Button transparent style={styles.buttonContainer} >
                    <Text style={styles.buttonLabelMobility}>No</Text>
                </Button>
            </View>

            <View style={styles.containerOfButtons3}>
                <View style={styles.labelContainer}>
                    <Text style={styles.shortLabel}>Method of Travel: </Text>
                </View>
                <Button transparent style={styles.buttonContainerMOT} >
                    <View style={styles.iconContainer}>
                        <Icon name="md-car" style={styles.icon}></Icon>
                    </View>
                    <View style={styles.iconLabelContainer}>
                        <Text style={styles.buttonLabel}>Car</Text>
                    </View>
                </Button>
                <Button transparent style={styles.buttonContainerMOT} >
                    <View style={styles.iconContainer}>
                        <Icon name="md-walk" style={styles.icon}></Icon>
                    </View>
                    <View style={styles.iconLabelContainer}>
                        <Text style={styles.buttonLabel}>Walking</Text>
                    </View>
                </Button>
                <Button transparent style={styles.buttonContainerMOT}>
                    <View style={styles.iconContainer}>
                        <Icon name="md-bus" style={styles.icon}></Icon>
                    </View>
                    <View style={styles.iconLabelContainer}>
                        <Text style={styles.buttonLabel}>Bus</Text>
                    </View>
                </Button>
                <Button transparent style={styles.buttonContainerMOT}>
                    <View style={styles.iconContainer}>
                        <Icon name="ios-bus" style={styles.icon}></Icon>
                    </View>
                    <View style={styles.iconLabelContainer}>
                        <Text style={styles.buttonLabel}>Shuttle Bus</Text>
                    </View>
                </Button>
            </View>

            <View style={styles.containerOfDisclaimer}>
                <View style={styles.labelContainer}>
                    <Text style={styles.shortLabel}>Disclaimer: </Text>
                </View>
                <View style={styles.disclaimerTextContainer}>
                    <Text style={styles.disclaimerText}>The Concordia shuttle bus offers you a free ride between the SGW and
                    Loyola campus. However, the services are reserved for students with valid ID cards
                        and buses are wheelchair accessible.</Text>
                </View>
            </View >

        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%"
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    shortLabel: {
        position: "absolute",
        color: "#FFFFFF",
        fontSize: 16,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
    },
    buttonContainer: {
        height: 80,
        width: 80,
        backgroundColor: "#353A50",
        borderRadius: 10,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        margin: "2%",
        top: "25%",
    },
    buttonContainerMOT: {
        height: 80,
        width: 80,
        backgroundColor: "#353A50",
        borderRadius: 10,
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
        margin: "2%",
        top: "8%",
    },
    buttonLabel: {
        position: "absolute",
        color: "#FFFFFF",
        fontSize: 10,
        fontFamily: "encodeSansExpanded",
        width: "100%",
        alignContent: "center",
        justifyContent: "center",
        textAlign: "center",
        paddingHorizontal: 10
    },
    buttonLabelMobility: {
        position: "absolute",
        color: "#FFFFFF",
        fontSize: 15,
        fontFamily: "encodeSansExpanded",
        width: "100%",
        alignContent: "center",
        justifyContent: "center",
        textAlign: "center"
    },
    labelContainer: {
        position: "absolute",
        left: "5%"
    },
    containerOfButtons1: {
        position: "absolute",
        width: "100%",
        height: "15%",
        flexDirection: "row",
        justifyContent: "center",
        bottom: "62%"
    },
    containerOfButtons2: {
        position: "absolute",
        width: "100%",
        height: "15%",
        flex: 1,
        flexDirection: "row",
        justifyContent: "center",
        bottom: "41%"
    },
    containerOfButtons3: {
        position: "absolute",
        width: "100%",
        height: "15%",
        flexDirection: "row",
        justifyContent: "center",
        bottom: "21%"
    },
    containerOfDisclaimer: {
        position: "absolute",
        width: "100%",
        height: "15%",
        flexDirection: "row",
        justifyContent: "center",
        bottom: "1%",
        flexShrink: 1,
    },
    disclaimerTextContainer: {
        position: "absolute",
        width: "90%",
        flexDirection: "row",
        justifyContent: "center",
        flexShrink: 1,
        bottom: "40%"
    },
    disclaimerText: {
        flex: 1,
        flexWrap: "wrap",
        color: "#919090",
        fontSize: 10,
        fontFamily: "encodeSansExpanded",
        flexShrink: 1,
        textAlign: "left"
    },
    icon: {
        position: "absolute",
        color: "#FFFFFF",
        alignSelf: "center",
        fontSize: 35
    },
    iconContainer: {
        width: "100%",
        height: "60%",
    },
    iconLabelContainer: {
        width: "100%",
        height: "25%",
        bottom: "5%"
    },
});

export { PreferenceMenu };