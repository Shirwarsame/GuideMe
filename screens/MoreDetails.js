import React from 'react';
import {View, Text, StyleSheet, Image} from 'react-native'
import { Icon, Button, Right } from 'native-base';


function MoreDetails (props) {
    return(
        <View>
         
            <Text style={styles.mainLabel}>{props.name}</Text>
            <Button style={styles.directionButton}><Text style={{ color: 'white' }}>Get Directions</Text></Button>
            
        </View>  


    );
}

export const styles = StyleSheet.create({


    mainLabel: {
        color: '#FFFFFF',
        // position: 'absolute',
        fontSize: 30,
        fontWeight: 'bold',
        fontFamily: 'encodeSansExpanded',
        top:'100%'
    },

    directionButton: {
        width: 300,
        height: 60,
        top:'160%',
        left: '50%',
        fontSize: 25,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#3ACCE1',
        borderRadius: 10,
    }

});


export { MoreDetails }