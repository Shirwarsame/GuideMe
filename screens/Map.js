/* eslint-disable no-unused-vars */
import React from "react";
import { StyleSheet, Keyboard  } from "react-native";
import MapView, { PROVIDER_GOOGLE } from "react-native-maps";
import BuildingHighlight  from "../components/BuildingHighlight";
import { BuildingIdentification } from "../components/BuildingIdentification";
import { BottomMenu } from "../components/BottomMenu";
import CurrentBuildingLocation  from "../components/CurrentBuildingLocation";
import CurrentLocationButton  from "../components/CurrentLocationButton";
import { View } from "native-base";
import Search  from "../components/Search";
import PropTypes from "prop-types";
import { LocationMarker } from "../components/locationMarker";
import {CampusRegion} from "../constants/buildingData";
import { connect } from "react-redux";


function mapStateToProps(state) {
    return {
        selectedBuildingName: state.selectedBuildingName
    }
}


function mapDispatchToProps(dispatch) {
    return {
        setSelectedBuildingName: (value) => dispatch({ type: "UPDATE_SELECTED_BUILDING", payload: value }),
    }
}


/**
 * US1 - As a user, I would like to navigate through SGW campus.
 * US2 - As a user, I would like to navigate through Loyola campus.
 * 
 * This is our main screen which includes all the components inside a map.
 */
export function Map (props) { 

    const [isMapClicked, setIsMapClicked] = React.useState(false);
    const mapRef = React.useRef(null);
    
    return (
        <View testID="mapView" data-test="MapComponent">
            <View>
                <MapView
                    id="mainMap"
                    data-test="MapViewComponent"
                    ref={mapRef}
                    style={styles.map}
                    provider={PROVIDER_GOOGLE}
                    initialRegion={CampusRegion.sgwCoord}
                    showsUserLocation={true}
                    showsCompass={true}
                    showsBuildings={true}
                    showsIndoors={false}
                    onPress = {() => { 
                        if(isMapClicked) {
                            setIsMapClicked(false);
                            Keyboard.dismiss();
                        }
                        else {
                            setIsMapClicked(true);
                            Keyboard.dismiss();
                            props.setSelectedBuildingName(null);
                        }
                    }}

                >
                    <BuildingHighlight />
                    <BuildingIdentification />
                    <LocationMarker mapReference = {mapRef} />
                </MapView>
                <Search testID="searchBar" 
                navigation = {props.navigation} 
                mapReference = {mapRef} 
                />
                <View style = {styles.overlayButtons}>
                        <CurrentBuildingLocation mapReference = {mapRef}/>
                        <View style = {styles.spaceBetweenOverlayedButtons}></View>
                        <CurrentLocationButton mapReference = {mapRef} />
                </View>
            <BottomMenu navigation={props.navigation} mapReference = {mapRef} />
            </View>
        </View>
    );
}

Map.propTypes = {
    navigation: PropTypes.any,
};



export const styles = StyleSheet.create({
    map: {
        height: "100%"
    },
   overlayButtons: {
    position: "absolute", 
    flexDirection: "column", 
    top:"70%", 
    left: "80%"
   },
   spaceBetweenOverlayedButtons: {
       height: "5%"
   }
});

export default connect(mapStateToProps, mapDispatchToProps)(Map);

