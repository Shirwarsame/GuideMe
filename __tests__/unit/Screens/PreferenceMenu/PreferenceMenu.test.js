import React from "react";
import renderer from "react-test-renderer";
import {PreferenceMenu} from "../../../../screens/PreferenceMenu";
import { shallow } from 'enzyme';
import { Icon, Button } from "native-base";


describe("Preference Menu Screen", () => {
    test("renders correctly", () => {
      const navigation = { getParam: (param, defaultValue) => {
          return "indoor";
        }};
      const tree = renderer.create(<PreferenceMenu navigation={navigation} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test("click on icon when indoor map should load up pref menu", () => {
      const navigation = {
        getParam: (param, defaultValue) => {
          return "indoor"
        },
        navigate: (param, defaultValue) => {
          return "Hall Building"
        }
      };
      const prefWrapper = shallow(<PreferenceMenu navigation={navigation} />);
      prefWrapper.find("#prefIcon").props().onPress();
    });

    test("click on icon when preview directions should load up pref menu", () => {
      const navigation = {
        getParam: (param, defaultValue) => {
          return null
        },
        navigate: (param, defaultValue) => {
          return "Hall Building"
        }
      };
      const prefWrapper = shallow(<PreferenceMenu navigation={navigation} />);
      prefWrapper.find("#prefIcon").props().onPress();
    });

    test("click on button when indoor map should load up pref menu", () => {
      const setPersonaType = (val) => {
        return val;
      }
      const setMobilityReduced = (val) => {
        return val;
      }
      const setMethodOfTravel = (val) => {
        return val;
      }
      const navigation = {
        getParam: (param, defaultValue) => {
          return "indoor"
        }
      };
      const prefWrapper = shallow(<PreferenceMenu navigation={navigation} setPersonaType={setPersonaType} setMobilityReduced={setMobilityReduced} setMethodOfTravel={setMethodOfTravel} />);
      prefWrapper.find("#personaBtn").props().onPress();
      prefWrapper.find("#personaBtnUnderGrad").props().onPress();
      prefWrapper.find("#personaBtnVisitor").props().onPress();
      prefWrapper.find("#personaBtnStaff").props().onPress();
      prefWrapper.find("#mobReduced").props().onPress();
      prefWrapper.find("#mobNotReduced").props().onPress();
      prefWrapper.find("#driving").props().onPress();
      prefWrapper.find("#driving").props().onPress();
      prefWrapper.find("#walking").props().onPress();
      prefWrapper.find("#walking").props().onPress();
      prefWrapper.find("#bicycle").props().onPress();
      prefWrapper.find("#bus").props().onPress();
    });
});
