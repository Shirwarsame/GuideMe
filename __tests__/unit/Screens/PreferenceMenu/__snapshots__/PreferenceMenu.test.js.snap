// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preference Menu Screen Should render without errors 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2A2E43",
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 25,
        "fontWeight": "bold",
        "position": "absolute",
        "top": "15%",
      }
    }
  >
    Preferences
  </Text>
  <View
    style={
      Object {
        "bottom": "62%",
        "flexDirection": "row",
        "height": "15%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": "5%",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 16,
            "fontWeight": "bold",
            "position": "absolute",
          }
        }
      >
        I am: 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "25%",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 10,
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        Graduate Student
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "25%",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 10,
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        Undergrad Student
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "25%",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 10,
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        Visitor
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "25%",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 10,
            "justifyContent": "center",
            "paddingHorizontal": 10,
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        University Staff
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": "41%",
        "flex": 1,
        "flexDirection": "row",
        "height": "15%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": "5%",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 16,
            "fontWeight": "bold",
            "position": "absolute",
          }
        }
      >
        Mobility Reduced: 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "25%",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 15,
            "justifyContent": "center",
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
         Yes 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "25%",
          "width": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 15,
            "justifyContent": "center",
            "position": "absolute",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        No
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": "21%",
        "flexDirection": "row",
        "height": "15%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": "5%",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 16,
            "fontWeight": "bold",
            "position": "absolute",
          }
        }
      >
        Method of Travel: 
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "column",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "8%",
          "width": 80,
        }
      }
    >
      <View
        style={
          Object {
            "height": "60%",
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "bottom": "5%",
            "height": "25%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignContent": "center",
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 10,
              "justifyContent": "center",
              "paddingHorizontal": 10,
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Car
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "column",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "8%",
          "width": 80,
        }
      }
    >
      <View
        style={
          Object {
            "height": "60%",
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "bottom": "5%",
            "height": "25%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignContent": "center",
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 10,
              "justifyContent": "center",
              "paddingHorizontal": 10,
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Walking
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "column",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "8%",
          "width": 80,
        }
      }
    >
      <View
        style={
          Object {
            "height": "60%",
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "bottom": "5%",
            "height": "25%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignContent": "center",
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 10,
              "justifyContent": "center",
              "paddingHorizontal": 10,
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Bicycling
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353A50",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 10,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "column",
          "height": 80,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": "8%",
          "width": 80,
        }
      }
    >
      <View
        style={
          Object {
            "height": "60%",
            "width": "100%",
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "bottom": "5%",
            "height": "25%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignContent": "center",
              "color": "#FFFFFF",
              "fontFamily": "encodeSansExpanded",
              "fontSize": 10,
              "justifyContent": "center",
              "paddingHorizontal": 10,
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Transit
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": "1%",
        "flexDirection": "row",
        "flexShrink": 1,
        "height": "15%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": "5%",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 16,
            "fontWeight": "bold",
            "position": "absolute",
          }
        }
      >
        Disclaimer: 
      </Text>
    </View>
    <View
      style={
        Object {
          "bottom": "20%",
          "flexDirection": "column",
          "flexShrink": 1,
          "justifyContent": "center",
          "position": "absolute",
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#919090",
            "flex": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 10,
            "textAlign": "left",
          }
        }
      >
        The Concordia shuttle bus offers you a free ride between the SGW and Loyola campus. However, the services are reserved for students with valid ID cards and buses are wheelchair accessible.
      </Text>
      <Text
        style={
          Object {
            "color": "#919090",
            "flex": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "fontFamily": "encodeSansExpanded",
            "fontSize": 10,
            "textAlign": "left",
          }
        }
      >
        The shuttle bus is only useful to travel between campuses.
      </Text>
      <Text
        style={
          Object {
            "color": "#919090",
            "flex": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "fontSize": 11,
            "fontWeight": "bold",
            "textAlign": "left",
          }
        }
      >
        Head to the Side Menu if you wish to see the next departures.
      </Text>
    </View>
  </View>
</View>
`;
